### PR TITLE
update generate ssl doc

### PR DESCRIPTION
**Pull request by hunterlxt**
### What is changed, added or deleted? (Required)

remove cfssl usage and switch to openssl

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### CODE REVIEW

File: generate-self-signed-certificates.md [hunk 1]
```diff
@@ -8,156 +8,108 @@ aliases: ['/docs-cn/dev/how-to/secure/generate-self-signed-certificates/']
 
 ## 概述
 
-本文档提供使用 `cfssl` 生成自签名证书的示例。
+本文档仅提供使用 `openssl` 生成自签名证书的一个示例，用户也可以根据自己的需求生成符合自己需求的证书和密钥。
```
Comment (yikeke) [inline generate-self-signed-certificates.md:11]:
```suggestion
本文档提供使用 `openssl` 生成自签名证书的一个示例，用户也可以根据自己的需求生成符合要求的证书和密钥。
```

File: generate-self-signed-certificates.md [hunk 2]
```diff
@@ -8,156 +8,108 @@ aliases: ['/docs-cn/dev/how-to/secure/generate-self-signed-certificates/']
...
 
-## 下载 cfssl
+## 安装 OpenSSL
 
-假设使用 x86_64 Linux 主机：
-
-{{< copyable "shell-regular" >}}
+对于 Debian 或 Ubuntu 操作系统：
 
 ```bash
```
Comment (yikeke) [inline generate-self-signed-certificates.md:28]:
```suggestion
{{< copyable "shell-regular" >}}

```bash
```

File: generate-self-signed-certificates.md [hunk 3]
```diff
@@ -8,156 +8,108 @@ aliases: ['/docs-cn/dev/how-to/secure/generate-self-signed-certificates/']
...
-export PATH=$PATH:~/bin
+apt install openssl
 ```
 
-## 初始化证书颁发机构
-
-生成 cfssl 的默认配置，以便于之后修改：
+对于 RedHat 或 CentOS 操作系统：
 
 ```bash
```
Comment (yikeke) [inline generate-self-signed-certificates.md:34]:
```suggestion
{{< copyable "shell-regular" >}}

```bash
```

File: generate-self-signed-certificates.md [hunk 4]
```diff
@@ -8,156 +8,108 @@ aliases: ['/docs-cn/dev/how-to/secure/generate-self-signed-certificates/']
...
-        {
-            "C": "CN",
-            "L": "Beijing",
-            "O": "PingCAP",
-            "ST": "Beijing"
-        }
-    ]
-}
-```
+也可以参考 OpenSSL 官方的 [下载文档](https://www.openssl.org/source/) 安装
```
Comment (yikeke) [inline generate-self-signed-certificates.md:38]:
```suggestion
也可以参考 OpenSSL 官方的[下载文档](https://www.openssl.org/source/) 进行安装。
```

File: generate-self-signed-certificates.md [hunk 5]
```diff
@@ -8,156 +8,108 @@ aliases: ['/docs-cn/dev/how-to/secure/generate-self-signed-certificates/']
 
 ## 概述
```
Comment (yikeke) [inline generate-self-signed-certificates.md:10]:
```suggestion
```

File: generate-self-signed-certificates.md [hunk 6]
```diff
@@ -8,156 +8,108 @@ aliases: ['/docs-cn/dev/how-to/secure/generate-self-signed-certificates/']
...
-    ]
-}
-```
+也可以参考 OpenSSL 官方的 [下载文档](https://www.openssl.org/source/) 安装
 
-### 生成 CA 证书
+## 生成 CA 证书
 
-{{< copyable "shell-regular" >}}
+CA 的作用是签发证书，在实际情况里，请联系你的管理员签发证书或者使用信任的 CA 机构。CA 会管理多个证书对，这里只需生成原始的一对证书：
```
Comment (yikeke) [inline generate-self-signed-certificates.md:42]:
```suggestion
CA 的作用是签发证书。实际情况中，请联系你的管理员签发证书或者使用信任的 CA 机构。CA 会管理多个证书对，这里只需生成原始的一对证书：
```

File: generate-self-signed-certificates.md [hunk 7]
```diff
@@ -8,156 +8,108 @@ aliases: ['/docs-cn/dev/how-to/secure/generate-self-signed-certificates/']
...
+### 集群中可能使用到的证书
 
-### 生成服务器端证书
+- tidb certificate 由 TiDB 使用，为其他组件和客户端验证 TiDB 身份。
+- tikv certificate 由 TiKV 使用，为其他组件和客户端验证 TiKV 身份。
+- pd certificate 由 PD 使用，为其他组件和客户端验证 PD 身份。
+- client certificate 用于 PD、TiKV、TiDB 验证客户端。例如 `pd-ctl`，`tikv-ctl` 等。
 
-`hostname` 中为各组件的 IP 地址，以及 `127.0.0.1`
+### 给 TiKV 实例签发证书
```
Comment (yikeke) [inline generate-self-signed-certificates.md:62]: 下面这些步骤建议用有序列表：1. 2. 3. 这样，方便用户阅读

**Comment by yikeke** [comment]
PTAL @Connor1996 @kissmydb

**Comment by hunterlxt** [comment]
Comments addressed, thank @yikeke for review

File: generate-self-signed-certificates.md [hunk 8]
```diff
@@ -6,158 +6,144 @@ aliases: ['/docs-cn/dev/how-to/secure/generate-self-signed-certificates/']
...
+
+7. 确认在当前目录下得到如下文件：
+
+    ```
+    root.crt
+    tikv.crt
+    tikv.key
+    ```
+
+为其它组件签发证书的过程类似，此文档不再赘述。
```
Comment (yikeke) [inline generate-self-signed-certificates.md:149]: See https://github.com/pingcap/docs-cn/pull/3259#discussion_r429723944. @hunterlxt

**Comment by hunterlxt** [comment]
@yikeke This PR is ready

File: generate-self-signed-certificates.md [hunk 9]
```diff
@@ -6,158 +6,170 @@ aliases: ['/docs-cn/dev/how-to/secure/generate-self-signed-certificates/']
...
+
+1. 生成该证书对应的私钥：
+
+    {{< copyable "shell-regular" >}}
+
+    ```bash
+    openssl genrsa -out client.key 2048
+    ```
+
+2. 生成证书请求文件（在这一步也可以为该证书指定 Common Name，其作用是让服务端验证接入的客户端的身份，各个组件默认不会开启验证，需要在配置文件中启用该功能才生效）
```
Comment (yikeke) [inline generate-self-signed-certificates.md:161]:
```suggestion
2. 生成证书请求文件（在这一步也可以为该证书指定 Common Name，其作用是让服务端验证接入的客户端的身份，默认不会开启对各个组件的验证，需要在配置文件中启用该功能才生效）
```

File: generate-self-signed-certificates.md [hunk 10]
```diff
@@ -6,158 +6,170 @@ aliases: ['/docs-cn/dev/how-to/secure/generate-self-signed-certificates/']
...
+
+    ```
+    root.crt
+    tikv.crt
+    tikv.key
+    ```
+
+为其它 TiDB 组件签发证书的过程类似，此文档不再赘述。
+
+### 为 Client 签发证书
```
Comment (yikeke) [inline generate-self-signed-certificates.md:151]:
```suggestion
### 为客户端签发证书

为客户端签发证书的步骤如下。
```

**Comment by hunterlxt** [comment]
> LGTM. And please make sure you have tried it manually.

@Connor1996 Actually, this document is my note for testing grpc-rs tls function :)

**Comment by hunterlxt** [comment]
@yikeke PTAL

**Pull request merged**